docs/zero_to_hero_guide/quickstart.md

```diff
@@ -0,0 +1,183 @@
+```
+
+
+### 2. Run the Llama Stack Distribution
+> Launching a distribution initializes and configures the necessary APIs and Providers, enabling seamless interaction with the underlying model.
+
+Start the server with the configured stack:
+
+```bash
+cd llama-stack/distributions/meta-reference-gpu
+llama stack run ./run.yaml
+```
+
+*The server will start and listen on `http://localhost:5000` by default.*
+
+---
+
+## Testing with `curl`
+
+After setting up the server, verify it's working by sending a `POST` request using `curl`:
+
+```bash
+curl http://localhost:5000/inference/chat_completion \
+-H "Content-Type: application/json" \
+-d '{
+    "model": "Llama3.1-8B-Instruct",
+    "messages": [
+        {"role": "system", "content": "You are a helpful assistant."},
+        {"role": "user", "content": "Write me a 2-sentence poem about the moon"}
+    ],
+    "sampling_params": {"temperature": 0.7, "seed": 42, "max_tokens": 512}
+}'
+```
+
+**Expected Output:**
+```json
+{
+  "completion_message": {
+    "role": "assistant",
+    "content": "The moon glows softly in the midnight sky,\nA beacon of wonder, as it catches the eye.",
+    "stop_reason": "out_of_tokens",
+    "tool_calls": []
+  },
+  "logprobs": null
+}
+```
+
+---
+
+## Testing with Python
+
+You can also interact with the Llama Stack server using a simple Python script. Below is an example:
+
+### 1. Install Required Python Packages
+The `llama-stack-client` library offers a robust and efficient python methods for interacting with the Llama Stack server.
+
+```bash
+pip install llama-stack-client
+```
+
+### 2. Create a Python Script (`test_llama_stack.py`)
+
+```python
+from llama_stack_client import LlamaStackClient
+from llama_stack_client.types import SystemMessage, UserMessage
+
+# Initialize the client
+client = LlamaStackClient(base_url="http://localhost:5000")
+
+# Create a chat completion request
+response = client.inference.chat_completion(
+    messages=[
+        SystemMessage(content="You are a helpful assistant.", role="system"),
+        UserMessage(content="Write me a 2-sentence poem about the moon", role="user")
+    ],
+    model="Llama3.1-8B-Instruct",
+)
+
+# Print the response
+print(response.completion_message.content)
+```
+
+### 3. Run the Python Script
+
+```bash
+python test_llama_stack.py
+```
+
+**Expected Output:**
+```
+The moon glows softly in the midnight sky,
+A beacon of wonder, as it catches the eye.
+```
+
+With these steps, you should have a functional Llama Stack setup capable of generating text using the specified model. For more detailed information and advanced configurations, refer to some of our documentation below.
+
+---
+
+## Next Steps
+
+**Explore Other Guides**: Dive deeper into specific topics by following these guides:
+- [Understanding Distribution](https://llama-stack.readthedocs.io/en/latest/getting_started/index.html#decide-your-inference-provider)
+- [Inference 101](00_Inference101.ipynb)
```

Comment:
For each of the notebooks here, can we 
1) add an "Open in Colab" link, which can be manually but accurately generated from the github repo notebook link (cf https://github.com/meta-llama/llama-recipes/blob/main/recipes/quickstart/Prompt_Engineering_with_Llama_3.ipynb);
2) show the use of the Together's llama stack server API (in addition to locally launched server) - example https://colab.research.google.com/drive/1AmdIPJG6ZqWVaB4zUTKfcZ8AvdnWX_3J#scrollTo=_Yc4a_abBCeh, for developers who can't install a local server or prefer to have a quick start.

Reply:
+1 on together

Reply:
Justin thanks for adding the Colab links - the branch name "docs_improvement" in the link e.g. https://colab.research.google.com/github/meta-llama/llama-stack/blob/docs_improvement/docs/zero_to_hero_guide/04_Tool_Calling101.ipynb works now, but after the merge and if the branch gets deleted, the links will be invalidate. So the links should better be `blob/main` instead of `blob/docs_improvement` even though using `main` doesn't work before the merge. @heyjustinai

Reply:
@jeffxtang thanks for pointing this out, changed the url. I agree to have a quick start for together as well, but will create a separate notebook for that, philosophy is to keep the guide simple and easy to follow